### PR TITLE
fix: use testify instead of t.Fatal or t.Error in client package (part 1)

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

This uses testify instead of testing for t.Fatal or t.Error calls in client package

Related to #18972

### CODE REVIEW

**Comment by ivanvc** [comment]
Question, @mmorel-35. How are you planning on splitting these pull requests? I want to understand it so that I can deliver a proper review :)

**Review by ivanvc — COMMENTED** [review]
Thanks for the work, @mmorel-35.
I did a grep in the `client` directory from this branch, and it looks like there are only 24 more instances of `t.Fatal` after your changes. If you add them, the PR will still be less than 200 changed lines.

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @mmorel-35

File: client/pkg/fileutil/dir_windows.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 
 const (
 	// PrivateDirMode grants owner to make/remove files inside the directory.
-	PrivateDirMode = 0777
+	PrivateDirMode = 0o777
```
[the remark this answers is not in the document]
Comment (jmhbnz) [reply client/pkg/fileutil/dir_windows.go:26]: I don't mind this being included, could add it as a separate commit perhaps but not a big deal.

**Review by ahrtr — APPROVED** [review]

**Pull request merged**